tests/check_linux_layout_intel.py

```diff
@@ -0,0 +1,192 @@
+    "sof/community/sof-glk.ri",
+    "sof/community/sof-adl.ri",
+    "sof/community/sof-rpl.ri",
+    "sof/community/sof-cnl.ri",
+    "sof/community/sof-tgl.ri",
+    "sof/community/sof-icl.ri",
+    "sof/community/sof-tgl-h.ri",
+    "sof/community/sof-rpl-s.ri",
+    "sof/community/sof-jsl.ri",
+    "sof/community/sof-adl-s.ri",
+    "sof/community/sof-cfl.ri",
+    "sof/community/sof-cml.ri",
+    "sof/intel-signed/sof-apl.ri",
+    "sof/intel-signed/sof-adl-n.ri",
+    "sof/intel-signed/sof-ehl.ri",
+    "sof/intel-signed/sof-glk.ri",
+    "sof/intel-signed/sof-adl.ri",
+    "sof/intel-signed/sof-rpl.ri",
+    "sof/intel-signed/sof-cnl.ri",
+    "sof/intel-signed/sof-tgl.ri",
+    "sof/intel-signed/sof-icl.ri",
+    "sof/intel-signed/sof-tgl-h.ri",
+    "sof/intel-signed/sof-rpl-s.ri",
+    "sof/intel-signed/sof-jsl.ri",
+    "sof/intel-signed/sof-adl-s.ri",
+    "sof/intel-signed/sof-cfl.ri",
+    "sof/intel-signed/sof-cml.ri",
+    "sof/sof-cnl.ri",
+    "sof/sof-cht.ri",
+    "sof/sof-tgl.ri",
+    "sof/sof-byt.ri",
+    "sof/sof-icl.ri",
+    "sof/sof-bdw.ri",
+    "sof/sof-tgl-h.ri",
+    "sof/sof-rpl-s.ri",
+    "sof/sof-jsl.ri",
+    "sof/sof-adl-s.ri",
+    "sof/sof-cfl.ri",
+    "sof/sof-cml.ri",
+    "sof-ipc4/rpl/sof-rpl.ri",
+    "sof-ipc4/rpl/community/sof-rpl.ri",
+    "sof-ipc4/rpl/intel-signed/sof-rpl.ri",
+    "sof-ipc4/adl-n/sof-adl-n.ri",
+    "sof-ipc4/adl-n/community/sof-adl-n.ri",
+    "sof-ipc4/adl-n/intel-signed/sof-adl-n.ri",
+    "sof-ipc4/tgl-h/community/sof-tgl-h.ri",
+    "sof-ipc4/tgl-h/intel-signed/sof-tgl-h.ri",
+    "sof-ipc4/tgl-h/sof-tgl-h.ri",
+    "sof-ipc4/mtl/community/sof-mtl.ri",
+    "sof-ipc4/mtl/intel-signed/sof-mtl.ri",
+    "sof-ipc4/mtl/sof-mtl.ri",
+    "sof-ipc4/rpl-s/community/sof-rpl-s.ri",
+    "sof-ipc4/rpl-s/intel-signed/sof-rpl-s.ri",
+    "sof-ipc4/rpl-s/sof-rpl-s.ri",
+    "sof-ipc4/adl-s/community/sof-adl-s.ri",
+    "sof-ipc4/adl-s/intel-signed/sof-adl-s.ri",
+    "sof-ipc4/adl-s/sof-adl-s.ri",
+    "sof-ipc4/tgl/community/sof-tgl.ri",
+    "sof-ipc4/tgl/intel-signed/sof-tgl.ri",
+    "sof-ipc4/tgl/sof-tgl.ri",
+    "sof-ipc4/adl/sof-adl.ri",
+    "sof-ipc4/adl/community/sof-adl.ri",
+    "sof-ipc4/adl/intel-signed/sof-adl.ri",
+ ]
+
+BASE="/lib/firmware/intel"
+
+def check_fw_files_kernel_to_bundle():
+    for i in fw_list:
+        full_path = bundle_base + "/" + i
+        if os.path.isfile(full_path):
+            if verbose_mode: print("File %s found." % full_path);
+        else:
+            raise Exception("File %s NOT found." % full_path);
+
+def check_fw_files_bundle_to_kernel():
+    #for entry in os.scandir(bundle_base):
+    paths = pathlib.Path(bundle_base).glob("sof*/*/*.ri")
+    for path in paths:
+        bundle_path  = path.relative_to(bundle_base)
+        if not str(bundle_path) in fw_list:
+            raise Exception("Firmware %s not expected by SOF Linux driver" % str(bundle_path))
+
+def check_sign_key():
+    for i in fw_list:
```

Comment:
I see two problems with this `for` loop:

1. It should be outside the `check_sign_key(file)` function, which should be focused on a single thing: checking a single file. Not just to avoid a lot of spurious indentation but also to separate the low level logic and the iteration; they're two different things. This will also make interactive testing easier.

2. This is hardcoding this useful function to a specific data set. It should be easy to point this check at any directory and check all (`.ri`) files there no matter which specific ones they are.